lib/EvalSourceMapDevToolPlugin.js

```diff
@@ -153,7 +154,15 @@ class EvalSourceMapDevToolPlugin {
 						);
 						sourceMap.sources = moduleFilenames;
 						sourceMap.sourceRoot = options.sourceRoot || "";
-						const moduleId = chunkGraph.getModuleId(m);
+						const moduleId = chunkGraph.getModuleId(m).toString();
+						const moduleIdExtension = extname(moduleId);
+						if (moduleIdExtension === "") {
```

Comment:
I think better to use regexp here, because it can be like pseudo URL (like `index.js?foo=bar`, yep, legacy stuff and etc)

Reply:
Oh, get it, I'll use it here 👌

Reply:
why not just use `sourceMap.file = moduleId`

Reply:
because as alexander pointed out, in some cases we have the extension and in other we don't

Reply:
I added utility function for getting the extension from different paths, including pseudo URL and tested it agains extname's test cases + 2 custom pseudo URL. Seems like it should work with all kind of possible paths now.

Reply:
I fixed the lint issue. It was a simple misorder.